apps/heft/src/plugins/CopyGlobsPlugin.ts

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
```

Comment:
Every single time a well-meaning glob like `"src/**/*.md"` appears in one of our config files, it will most likely be processed as an inefficient `O(n)` brute force scan of the folder tree.  

- Can we call this out in the docs?  
- Can we design the tooling in a way that minimizes usage of globs?
- If not, could we invent some kind of fancy file cache that performs the expensive disk scan only once, and then matches globs against an efficient index?

Reply:
We have the `CachedFileSystem` type I believe in Rush, however I'm not sure using that here is a fantastic plan. We could alternatively allow for specifying exact folders/file paths that need to be copied within a project. We could also do a mixed approach, where we allow globs to select files but also allow for copying directories at exact locations. This would allow us the flexibility of regex when needed and the perf of copying specific directories when we know exactly where to look.

Reply:
It would be useful for this PR to more clearly describe some real world problems that "copyFiles" is meant to solve.  Do any of them actually require globs at all? Hard to say without knowing the specifics.

Reply:
I've thrown some more info into the summary